async.el

```diff
@@ -358,7 +376,8 @@ working directory."
     (with-current-buffer buf
       (set (make-local-variable 'async-callback) finish-func)
       (set-process-sentinel proc #'async-when-done)
-      (set-process-filter proc #'async-read-from-client)
+      (set-process-filter proc (async--process-filter-line-buffering-decorator
+                                #'async-read-from-client))
```

Comment:
This is more commonly done using `add-function`, see for example `M-x find-function RET shell-command RET C-M-e`.
```el
(add-function :around (process-filter proc) (lambda (filter proc output) ...))
```
Maybe you could even use `:before-until` or `:before-while` instead of `:around`?

Reply:
Honestly this seems rather complicated :O Does it really do the same thing?  I'm using this factory to get a closure with the `data` "buffer".

Reply:
> Honestly this seems rather complicated :O 

Does the intro in [`(info "(elisp) Advising Functions")`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Advising-Functions.html) help demysticise it at all?

> Does it really do the same thing?

Pretty much, but in a more flexible and introspectable way.

> I'm using this factory to get a closure with the `data` "buffer".

You can make a similar closure with the `:around` advice if you want to.

Reply:
I'll check the info page, it sounds interesting.  I'm only using advices through `add-advice` which is probably a short hand for something else.

In the meantime, to fix #169 I rewrote the code using a marker as @thierryvolpiatto suggested.

Reply:
`advice-add` is implemented in terms of `add-function`.
The former is for named function symbols, whereas the latter works on a variety of generalised variable places, such as `process-filter`.